src/compiler/config/config-utils.ts

```diff
@@ -21,64 +44,31 @@ export const setBooleanConfig = (config: any, configName: string, flagName: stri
     config[userConfigName] = !!config[userConfigName]();
   }
 
-  if (typeof config[userConfigName] === 'boolean') {
+  if (isBoolean(config[userConfigName])) {
     config[configName] = config[userConfigName];
   } else {
     config[configName] = defaultValue;
   }
 };
 
-export const setNumberConfig = (config: any, configName: string, _flagName: string, defaultValue: number) => {
-  const userConfigName = getUserConfigName(config, configName);
-
-  if (typeof config[userConfigName] === 'function') {
-    config[userConfigName] = config[userConfigName]();
-  }
-
-  if (typeof config[userConfigName] === 'number') {
-    config[configName] = config[userConfigName];
-  } else {
-    config[configName] = defaultValue;
-  }
-};
-
-export const setStringConfig = (config: any, configName: string, defaultValue: string) => {
-  const userConfigName = getUserConfigName(config, configName);
-
-  if (typeof config[userConfigName] === 'function') {
-    config[userConfigName] = config[userConfigName]();
-  }
-
-  if (typeof config[userConfigName] === 'string') {
-    config[configName] = config[userConfigName];
-  } else {
-    config[configName] = defaultValue;
-  }
-};
-
-export const setArrayConfig = (config: any, configName: string, defaultValue?: any[]) => {
-  const userConfigName = getUserConfigName(config, configName);
-
-  if (typeof config[userConfigName] === 'function') {
-    config[userConfigName] = config[userConfigName]();
-  }
-
-  if (!Array.isArray(config[configName])) {
-    if (Array.isArray(defaultValue)) {
-      config[configName] = defaultValue.slice();
-    } else {
-      config[configName] = [];
-    }
-  }
-};
-
-const getUserConfigName = (config: d.Config, correctConfigName: string) => {
+/**
+ * Find any possibly mis-capitalized configuration names on the config, logging
+ * a little warning for the user to let them know. This lets us recover values
+ * set under (a subset of) improperly spelled configs and automatically hoist
+ * them into the config under the right key.
+ *
+ * @param config d.Config
+ * @param correctConfigName the configuration name that we're checking for right now
+ * @returns a string container a mis-capitalized config name found on the
+ * config object, if any.
+ */
```

Comment:
this, like all other JSDoc comments I'm adding, reflects _my belief_ of what this function does - very possibly I am wrong!